toolchain/sem_ir/type_info.h

```diff
@@ -71,6 +74,61 @@ struct CompleteTypeInfo : public Printable<CompleteTypeInfo> {
   ValueRepr value_repr = ValueRepr();
 };
 
+// The initializing representation to use when returning by value.
+struct InitRepr {
+  enum Kind : int8_t {
+    // The type has no initializing representation. This is used for empty
+    // types, where no initialization is necessary.
+    None,
+    // An initializing expression produces an object representation by value,
+    // which is copied into the initialized object.
+    ByCopy,
+    // An initializing expression takes a location as input, which is
+    // initialized as a side effect of evaluating the expression.
+    InPlace,
+    // No initializing expressions should exist because the type is not
+    // complete.
+    Incomplete,
+    // TODO: Consider adding a kind where the expression takes an advisory
+    // location and returns a value plus an indicator of whether the location
+    // was actually initialized.
+  };
+  // The kind of initializing representation used by this type.
+  Kind kind;
+
+  // Returns whether the initializing representation information could be fully
+  // computed.
+  auto is_valid() const -> bool { return kind != Incomplete; }
+};
+
+// Returns information about the initializing representation to use for a type.
+auto GetInitRepr(const File& file, TypeId type_id) -> InitRepr;
```

Comment:
Should this be InitRepr::ForType similar to below? (plus, member functions for factories makes sense to me)

Reply:
Done.